test/unit/event/send-batch-error-event.test.js

```diff
@@ -13,39 +42,103 @@ const error = {
+
+  event = {
+    name: 'batch-processing-error',
+    type: 'error'
+  }
+})
 
-    event = {
-      name: 'batch-processing-error',
-      type: 'error'
-    }
+afterEach(async () => {
+  jest.clearAllMocks()
+})
+
+describe('V1 send batch error event for SITI payment file that cannot be parsed', () => {
+  test('when V1 events enabled should call mockSendEvent when a filename and error is received', async () => {
+    processingConfig.useV1Events = true
+    await sendBatchErrorEvent(filename, error)
+    expect(mockSendEvent).toHaveBeenCalled()
   })
 
-  afterEach(async () => {
-    jest.resetAllMocks()
+  test('when V1 events disabled should not call mockSendEvent when a filename and error is received', async () => {
+    processingConfig.useV1Events = false
+    await sendBatchErrorEvent(filename, error)
+    expect(mockSendEvent).not.toHaveBeenCalled()
   })
 
   test('should call uuidv4 when a filename and error is received', async () => {
     await sendBatchErrorEvent(filename, error)
     expect(uuidv4).toHaveBeenCalled()
   })
 
-  test('should call raiseEvent when a filename and error is received', async () => {
+  test('should raise event with batch-processing-error event name', async () => {
     await sendBatchErrorEvent(filename, error)
-    expect(raiseEvent).toHaveBeenCalled()
+    expect(mockSendEvent.mock.calls[0][0].name).toBe(BATCH_PROCESSING_ERROR)
   })
 
-  test('should call raiseEvent with event and "error" when a filename and error is received', async () => {
-    event = {
-      ...event,
-      id: uuidv4(),
-      message: error.message,
-      data: { filename }
-    }
+  test('should raise event with error status', async () => {
+    await sendBatchErrorEvent(filename, error)
+    expect(mockSendEvent.mock.calls[0][0].properties.status).toBe('error')
+  })
+
+  test('should raise error event type', async () => {
+    await sendBatchErrorEvent(filename, error)
+    expect(mockSendEvent.mock.calls[0][0].properties.action.type).toBe(event.type)
+  })
+
+  test('should include error message in event', async () => {
+    await sendBatchErrorEvent(filename, error)
+    expect(mockSendEvent.mock.calls[0][0].properties.action.message).toBe(error.message)
+  })
+
+  test('should throw error if no error provided', async () => {
+    await expect(() => sendBatchErrorEvent(filename)).rejects.toThrow()
+  })
+})
+
+describe('V2 send batch error event for SITI payment file that cannot be parsed', () => {
+  test('send V2 events when v2 events enabled ', async () => {
+    processingConfig.useV2Events = true
+    await sendBatchErrorEvent(filename, error)
+    expect(mockPublishEvent).toHaveBeenCalled()
+  })
+
+  test('should not send V2 events when v2 events disabled ', async () => {
+    processingConfig.useV2Events = false
+    await sendBatchErrorEvent(filename, error)
+    expect(mockPublishEvent).not.toHaveBeenCalled()
+  })
+
+  test('should send event to V2 topic', async () => {
```

Comment:
do we need an equivalent test for sending to the V1 topic? Only ask as i tried to implement that as a test and couldnt get it to work and wondered why you removed it, is it not possible/easy/worth testing?

Reply:
My mistake, took it out to sort the other tests and forgot to re-add it.  Restored now.